homework-solutions/js/hw-4/task-3.js

```diff
@@ -6,3 +7,22 @@
   преобразовываясь в number
 
 */
+
+const minAge = 18; // step 1
+const maxAge = 60; // step 2
+let age = '20'; //step 3
+if (typeof +age != 'number'){ // start step 4-1
+  console.log('Incorrect data type')
+} // end step 4-1
+else if (age < minAge){
+  console.log('You don\'t have access cause your age is '+ age + '.' +' It\'s less than '+ minAge) //start step 4-2
```

Comment:
Вьіполняем проверку, если не подходит одно действие, если подходит - приводим к числу и проходим по ветвлениям
Неплохо узнать разницу между `isNaN` и `Number.isNaN`

В твоем варианте `+age` можем получить `NaN`, что является `number`

```suggestion
if (isNaN(age)) {
  console.log('Incorrect data type');
} else {
  age = +age; // Number(age)
  if (age < minAge) {
  console.log('You don\'t have access cause your age is '+ age + '.' +' It\'s less than '+ minAge) //start step 4-2
```

Reply:
Принял к размышлениям.